pkg/collector/corechecks/containers/docker_rate_capping.go

```diff
@@ -0,0 +1,125 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017 Datadog, Inc.
+
+// +build docker
+
+package containers
+
+import (
+	"sort"
+	"time"
+
+	log "github.com/cihub/seelog"
+
+	"github.com/DataDog/datadog-agent/pkg/aggregator"
+	"github.com/DataDog/datadog-agent/pkg/util/cache"
+)
+
+/*
+ * Rate capping logic to work around buggy kernel versions reporting artifical
+ * spikes in the cpuacct metrics. This logic is specific to this check and not
+ * ported downstream in the aggregator as it's not meant as a "standard" feature.
+ *
+ * This is triggered by users uncommenting the `capped_metrics` section in docker.yaml
+ */
+
+const dockerRateCacheKey = "docker_rate_prev_value"
+
+// cappedSender wraps around the standard Sender and overrides
+// the Rate method to implement rate capping
+type cappedSender struct {
+	aggregator.Sender
+	rateCaps  map[string]float64
+	timestamp time.Time // Current time at check Run()
```

Comment:
Added here instead of checking `time.Now()` for performance + testability (see `GetSender` for the update of that value)